tests/playwright/src/model/ai-lab-page.ts

```diff
@@ -0,0 +1,132 @@
+  readonly heading: Locator;
+  readonly navigationBar: Locator;
+
+  readonly recipesCatalogButton: Locator;
+  readonly runningAppsButton: Locator;
+  readonly catalogButton: Locator;
+  readonly servicesButton: Locator;
+  readonly playgroundsButton: Locator;
+
+  readonly recipesCatalogPage: Locator;
+  readonly recipesCatalogContent: Locator;
+  readonly recipesCatalogNaturalLanguageProcessing: Locator;
+  readonly recipesCatalogAudio: Locator;
+  readonly recipesCatalogComputerVision: Locator;
+
+  readonly applicationDetailsPanel: Locator;
+  readonly startAIAppButton: Locator;
+  readonly openAIAppButton: Locator;
+  readonly deleteAIAppButton: Locator;
+
+  constructor(page: Page, webview: Page) {
+    this.page = page;
+    this.webview = webview;
+    this.heading = webview.getByLabel('Welcome to Podman AI Lab');
+    this.navigationBar = webview.getByLabel('PreferencesNavigation');
+
+    this.recipesCatalogButton = this.navigationBar.getByLabel('Recipes Catalog', { exact: true });
+    this.runningAppsButton = this.navigationBar.getByLabel('Running');
+    this.catalogButton = this.navigationBar.getByLabel('Catalog', { exact: true });
+    this.servicesButton = this.navigationBar.getByLabel('Services');
+    this.playgroundsButton = this.navigationBar.getByLabel('Playgrounds');
+
+    this.recipesCatalogPage = webview.getByRole('region', { name: 'Recipe Catalog' })
+    this.recipesCatalogContent = this.recipesCatalogPage.getByRole('region', { name: 'content', exact: true }).first();
+    this.recipesCatalogNaturalLanguageProcessing = this.recipesCatalogContent.getByRole('region', { name: 'Natural Language Processing', exact: true });
+    this.recipesCatalogAudio = this.recipesCatalogContent.getByRole('region', { name: 'Audio', exact: true });
+    this.recipesCatalogComputerVision = this.recipesCatalogContent.getByRole('region', { name: 'Computer Vision', exact: true });
+
+    this.applicationDetailsPanel = webview.getByLabel('application details panel');
+    this.startAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Start AI App' });
+    this.openAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Open AI App' });
+    this.deleteAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Delete AI App' });
+  }
+
+  async waitForPageLoad(): Promise<void> {
+    await playExpect(this.heading).toBeVisible();
+    await playExpect(this.navigationBar).toBeVisible();
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await playExpect(this.runningAppsButton).toBeVisible();
+    await playExpect(this.catalogButton).toBeVisible();
+    await playExpect(this.servicesButton).toBeVisible();
+    await playExpect(this.playgroundsButton).toBeVisible();
+  }
+
+  async openRecipesCatalog(): Promise<void> {
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await this.recipesCatalogButton.click();
+    await this.page.waitForTimeout(2000);
```

Comment:
We could again, wait for some locator count ie. `naturalLanguageProcessing.count() >= 0`. And one or two are enough? Ir does they get rendered sequentially?

Reply:
I agree, this should be changed to something more robust, if this timeout is required here then it should be changed to something that has more flexibility in case the environment is running slower or faster.

Reply:
updated in new page fragment